lib/main_common.pm

```diff
@@ -2616,6 +2616,14 @@ sub load_publiccloud_tests {
     if (get_var('PUBLIC_CLOUD_PREPARE_TOOLS')) {
         loadtest "publiccloud/prepare_tools";
     }
+    elsif (get_var('PUBLIC_CLOUD_QAM')) {
+        my $args = OpenQA::Test::RunArgs->new();
+        loadtest "publiccloud/ssh_init", run_args => $args;
```

Comment:
you just init args but not passing any values there so why actually you need them ?

Reply:
I am using it on `tests/publiccloud/ssh_interactive_init.pm`:
```
$args->{my_provider} = $provider;
$args->{my_instance} = $instance;
```
And then I am using this on `tests/publiccloud/ssh_interactive_end.pm` as:
```
$args->{my_provider}->cleanup(); 
```

Reply:
@asmorodskyi is this all right?

Reply:
I think it is actually expected way to use this feature .